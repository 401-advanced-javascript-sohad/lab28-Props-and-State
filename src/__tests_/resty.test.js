/* eslint-disable no-undefined */
/* eslint-disable no-unused-vars */
import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../components/main/form.js';
import Results from '../components/main/result.js';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Form />', ()=>{


  it('exist Form in our app', () => {
    let app = shallow(<Form />);
    expect(app.find('button').exists()).toBeTruthy();
  });

  it('correctly render to the DOM', ()=> {
    let app = renderer.create(<Form />);
    expect(app).toMatchSnapshot();
  });

  it('start of our application',()=>{
    let app = shallow(<Form />);
    expect(app.find('button').exists()).toBeTruthy();
    expect(app.find('input').exists()).toBeTruthy();
    expect(app.find('form').exists()).toBeTruthy();

  });
 
  it('renders correctly', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});