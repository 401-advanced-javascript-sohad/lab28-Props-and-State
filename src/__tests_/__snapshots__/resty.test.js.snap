// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form />  correctly to the DOM 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="useInput"
    id="input1"
    name="url"
    onChange={[Function]}
    placeholder="Type the URL  for the API"
    type="text"
  />
  <br />
  <button
    id="b1"
    name="GET"
    required={true}
  >
     GET
  </button>
  <button
    id="b1"
    name="POST"
  >
     POST
  </button>
  <button
    id="b1"
    tname="PUT"
  >
     PUT
  </button>
  <button
    className="PATCH"
    id="b1"
  >
     PATCH
  </button>
  <button
    id="b1"
    name="DELETE"
  >
     DELETE
  </button>
  <button
    id="go"
    type="submit"
  >
    Gooo!
  </button>
</form>
`;

exports[`<Form /> correctly render to the DOM 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="useInput"
    id="input1"
    name="url"
    onChange={[Function]}
    placeholder="Type the URL  for the API"
    type="text"
  />
  <br />
  <button
    id="b1"
    name="GET"
    required={true}
  >
     GET
  </button>
  <button
    id="b1"
    name="POST"
  >
     POST
  </button>
  <button
    id="b1"
    tname="PUT"
  >
     PUT
  </button>
  <button
    className="PATCH"
    id="b1"
  >
     PATCH
  </button>
  <button
    id="b1"
    name="DELETE"
  >
     DELETE
  </button>
  <button
    id="go"
    type="submit"
  >
    Gooo!
  </button>
</form>
`;

exports[`<Form /> render correctly to the DOM 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="useInput"
    id="input1"
    name="url"
    onChange={[Function]}
    placeholder="Type the URL  for the API"
    type="text"
  />
  <br />
  <button
    id="b1"
    name="GET"
    required={true}
  >
     GET
  </button>
  <button
    id="b1"
    name="POST"
  >
     POST
  </button>
  <button
    id="b1"
    tname="PUT"
  >
     PUT
  </button>
  <button
    className="PATCH"
    id="b1"
  >
     PATCH
  </button>
  <button
    id="b1"
    name="DELETE"
  >
     DELETE
  </button>
  <button
    id="go"
    type="submit"
  >
    Gooo!
  </button>
</form>
`;

exports[`<Form /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="useInput"
    id="input1"
    name="url"
    onChange={[Function]}
    placeholder="Type the URL  for the API"
    type="text"
  />
  <br />
  <button
    id="b1"
    name="GET"
    required={true}
  >
     GET
  </button>
  <button
    id="b1"
    name="POST"
  >
     POST
  </button>
  <button
    id="b1"
    tname="PUT"
  >
     PUT
  </button>
  <button
    className="PATCH"
    id="b1"
  >
     PATCH
  </button>
  <button
    id="b1"
    name="DELETE"
  >
     DELETE
  </button>
  <button
    id="go"
    type="submit"
  >
    Gooo!
  </button>
</form>
`;
